File: 0506-relative-ranks/0506-relative-ranks.cpp
#define pp pair<int,int>
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
          int n=score.size();
          priority_queue<pp> pq;
          for(int i=0;i<n;i++){
            pq.push({score[i],i});
          }
        int r=1;
         vector<string> ans(n);
         while(pq.size()>0){
            int idx = pq.top().second;
            pq.pop();
            if(r==1)  ans[idx]="Gold Medal";
            else if(r==2) ans[idx]="Silver Medal";
             else if(r==3)  ans[idx]="Bronze Medal";
            else      ans[idx]=to_string(r);
            r++;
         }
           
           return ans;
    }
};

File: 0506-relative-ranks/0506-relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
         int n = score.length;
        String[] result = new String[n];
        
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> b[0] - a[0]); // max-heap
        
        for (int i = 0; i < n; i++) {
            pq.offer(new int[]{score[i], i});
        }
        
        int rank = 1;
        
        while (!pq.isEmpty()) {
            int[] athlete = pq.poll();
            int idx = athlete[1];
            
            if (rank == 1) {
                result[idx] = "Gold Medal";
            } else if (rank == 2) {
                result[idx] = "Silver Medal";
            } else if (rank == 3) {
                result[idx] = "Bronze Medal";
            } else {
                result[idx] = String.valueOf(rank);
            }
            
            rank++;
        }
        
        return result;
    }
}